---
title: 编译器警告（等级 5）CS8892
ms.date: 08/26/2020
f1_keywords:
- CS8892
helpviewer_keywords:
- CS8892
author: Youssef1313
ms.openlocfilehash: a8568f96f5f21d4524280f88c76577a792a9f81d
ms.sourcegitcommit: d579fb5e4b46745fd0f1f8874c94c6469ce58604
ms.translationtype: HT
ms.contentlocale: zh-CN
ms.lasthandoff: 08/30/2020
ms.locfileid: "89147715"
---
# <a name="compiler-warning-level-5-cs8892"></a>编译器警告（等级 5）CS8892

方法“method”将不会用作入口点，因为找到了同步入口点“method”。

如果有多个有效入口点，其中包含一个或多个同步入口点以及一个或多个异步入口点，则将在所有异步入口点候选项上生成此警告。

由于仅从 C# 7.1 开始支持异步 main 方法，因此不会针对早期版本的项目生成此警告。

> [!NOTE]
> 编译器始终使用同步入口点。 如果有多个同步入口点，则会出现编译器错误。

## <a name="example"></a>示例

下面的示例生成 CS8892：

```csharp
using System;
using System.Threading.Tasks;

public class Program
{
    // CS8892: Method 'Program.Main()' will not be used as an entry point because a synchronous entry point 'Program.Main(string[])' was found.
    public static async Task Main()
    {
        await Task.Delay(1);
    }

    public static void Main(string[] args)
    {
        Console.WriteLine(2);
    }
}
```

## <a name="how-to-fix-this-warning"></a>如何修复此警告

仅保留程序的预期入口点，并重命名其他入口点。

```csharp
using System;
using System.Threading.Tasks;

public class Program
{
    public static async Task SomeOtherNameAsync()
    {
        await Task.Delay(1);
    }

    public static void Main(string[] args)
    {
        Console.WriteLine(2);
    }
}
```
